TLM/CSUtil.Commons/Log.cs

```diff
@@ -1,73 +1,78 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Diagnostics;
-using System.IO;
-using System.Threading;
-using UnityEngine;
+﻿namespace CSUtil.Commons {
+    using System;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Threading;
+    using UnityEngine;
 
-namespace CSUtil.Commons {
+    public static class Log {
+        private static readonly object LogLock_ = new object();
 
-	public static class Log {
-		private enum LogLevel {
-			Trace,
-			Debug,
-			Info,
-			Warning,
-			Error
-		}
+        // TODO refactor log filename to configuration
+        private static readonly string LogFilename_
+            = Path.Combine(Application.dataPath, "TMPE.log");
 
-		private static object logLock = new object();
+        private enum LogLevel {
+            Trace,
+            Debug,
+            Info,
```

Comment:
What's the difference between `Trace` and `Info`? Or, more specifically, which build do we have a conditional of `TRACE` but not `DEBUG`?

Reply:
It was not me who added TRACE level, it comes from the Harmony branch. From my experience TRACE is very detailed log to observe the execution flow, even more detailed than DEBUG.